### PR TITLE
Issue #135: Remove test dependency "Leo" and use PHPUnit exclusively.

### LINKED ISSUE #135: Why Peridot\\Leo assertions?
A lot of the tests use assertions from `Period\\Leo`, instead of native PhpUnit assertions.

E.g.

        expect($actual)->to->equal($item['return']);

instead of

        self::assertSame($item['return'], $actual);

I see no benefit here.
Let's use the more common/standard/familiar solution, if there is no good reason to use something else.

### PR DESCRIPTION
**Pull request by drupol**
Fix #135

### CODE REVIEW

**Review by richardcanoe — APPROVED** [review]

File: atomium/includes/common.inc [hunk 1]
```diff
@@ -318,7 +318,7 @@ function atomium_get_component_definitions() {
         }
 
         // Disable the preview if we explicitely set the key disable to TRUE.
-        if (TRUE == $definition['disable']) {
+        if (TRUE === $definition['disable']) {
```
Comment (donquixote) [inline atomium/includes/common.inc]: This "code style change" is not really neutral to behavior.
The value in `$definition['disable']` could be coming from a function `THEMENAME_atomium_definition_COMPONENTNAME()`, which is out of our control, and could set it to something like '1' or numeric 1 or 'true' or whatever, which previously was interpreted as TRUE.

At least we know due to the array + above that the array key is set.

The behavior change can be ok, I would personally put in a dedicated commit with clear message. But up to you.

File: atomium/includes/common.inc [hunk 2]
```diff
@@ -387,7 +387,7 @@ function atomium_menu_get_active_breadcrumb() {
 
     // Don't show a link to the current page in the breadcrumb trail.
     $end = \end($active_trail);
-    if ($item['href'] == $end['href']) {
+    if ($item['href'] === $end['href']) {
```
Comment (donquixote) [inline atomium/includes/common.inc]: This change is safe if we are comparing two strings.
Both values are provided and manipulated by external code, so in theory could be anything.

Again, this should perhaps be a dedicated commit with a message that indicates potential behavior change.

File: atomium/includes/common.inc [hunk 3]
```diff
@@ -404,7 +404,7 @@ function atomium_menu_get_active_breadcrumb() {
     // (commonly MENU_DEFAULT_LOCAL_TASK). In order to also hide the parent link
     // itself, we always remove the last link in the trail, if the current
     // router item links to its parent.
-    if (($item['type'] & MENU_LINKS_TO_PARENT) == MENU_LINKS_TO_PARENT) {
+    if (($item['type'] & MENU_LINKS_TO_PARENT) === MENU_LINKS_TO_PARENT) {
```
Comment (donquixote) [inline atomium/includes/common.inc]: I think this one is safe! The binary & will only return integer and the constants are integer, so yeah!

File: tests/fixtures/attributes/attributes.yml [hunk 4]
```diff
@@ -1,67 +1,191 @@
...
+    class: ['one', 'two']
+  run:
+    - append: ['class', 'three']
+    - remove: ['class', 'two']
+  expect:
+    - toArray:
+        return:
+          class: ['one', 'three']
+    - __toString:
+        return: ' class="one three"'
```
Comment (donquixote) [inline tests/fixtures/attributes/attributes.yml:12]: I don't know what these fixture changes have to do with the Leo / peridot assertions.
I imagine they will conflict with the Attributes PR.

Comment (drupol) [reply tests/fixtures/attributes/attributes.yml:12]: It's just a file rename.

**Comment by donquixote** [comment]
As discussed with @drupol in chat:

This PR in its current state contains a lot of changes that are unrelated to the goal stated in the PR title. Many of these changes will create conflicts with other PRs already open.

Let's reduce this PR to the changes that are actually required to remove the peridot/leo assertions.

Changes like this should be the essence of the PR:

```
-    expect($container['foo']->toArray())->to->be->equal(array('class' => array('bar')));
-    expect($container['bar']->toArray())->to->be->equal(array('class' => array('foo')));
+    self::assertEquals(array('class' => array('bar')), $container['foo']->toArray());
+    self::assertEquals(array('class' => array('foo')), $container['bar']->toArray());
```

All the other stuff should happen elsewhere, imo.

**Review by imanoleguskiza — CHANGES_REQUESTED** [review]
Just leaving a change request here to keep track of the comments above.

**Comment by drupol** [comment]
I just rebased the branch and included only the necessary changes related to the issue.

**Review by imanoleguskiza — APPROVED** [review]

**Comment by drupol** [comment]
Cherry-picked in 7.x-2.x as well.
